### PR TITLE
Add support `Karras sigmas` for StableDiffusionKDiffusionPipeline

### PR DESCRIPTION
**Pull request by takuma104**
This PR adds an option `use_karras_sigmas=` to apply the `Karras sigmas` discussed in issues like #1633, #2801 for the `StableDiffusionKDiffusionPipeline`. The original code is [mostly by @Stax124](https://github.com/Stax124/voltaML-fast-stable-diffusion/blob/4b6a23c5a09ee6375b6dbc86ba1d207df80eea7b/core/diffusers/kdiffusion/KDiffusionModel.py#L513), with some modifications made to apply it to the latest `StableDiffusionKDiffusionPipeline`. I opened this PR because the topic of `DPM++ 2M Karras` has come up repeatedly, and I personally wanted to use it.

## Usage:

```py
import torch
from diffusers import StableDiffusionKDiffusionPipeline
pipe = StableDiffusionKDiffusionPipeline.from_pretrained('dreamlike-art/dreamlike-anime-1.0', 
    torch_dtype=torch.float16,
    safety_checker = None,
).to('cuda')
pipe.set_scheduler('sample_dpmpp_2m')
image = pipe(prompt='best quality, a tuxedo cat',
    num_inference_steps=15, generator=torch.Generator(device='cuda').manual_seed(0), 
    width=512, height=512, 
    use_karras_sigmas=True).images[0]
image
```

## Effectiveness:

The sigmas seem to change slightly like this.

*Default:*
```py
[14.6146,  9.6826,  6.6780,  4.7746,  3.5221,  2.6666,  2.0606,  1.6156, 1.2768,  1.0097,  0.7913,  0.6056,  0.4397,  0.2780,  0.0292,  0.0000]
```

*Use Karras option:*
```py
[14.6146, 10.8198,  7.9029,  5.6878,  4.0277,  2.8016,  1.9104,  1.2741, 0.8289,  0.5243,  0.3211,  0.1896,  0.1072,  0.0576,  0.0292,  0.0000]
```

## Test: 
The integration tests have passed ~~(although there is no coverage for the additional code)~~. For generation test, I couldn't verify pixel matching, but the results seem to be close to those of `DPM++ 2M Karras` in stable-diffusion-webui.

|stable-diffusion-webui with `DPM++ 2M Karras` option |`use_karras_sigmas=True`|`use_karras_sigmas=False`|
|---|---|---|
|webui_00009-0-best quality, a tuxedo|diffusers_use_karras|diffusers_no_use_karras|

### CODE REVIEW

**Comment by patrickvonplaten** [comment]
Very clean! 

Could you just run `make style` to merge this one @takuma104 ? :-)

**Review by patrickvonplaten — APPROVED** [review]

**Comment by takuma104** [comment]
@patrickvonplaten thanks! I just fixed it.

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_k_diffusion.py [hunk 1]
```diff
@@ -456,7 +458,10 @@ def __call__(
             callback_steps (`int`, *optional*, defaults to 1):
                 The frequency at which the `callback` function will be called. If not specified, the callback will be
                 called at every step.
-
+            use_karras_sigmas (`bool`, *optional*, defaults to `False`):
+                Use karras sigmas. For example, specifying `sample_dpmpp_2m` to `set_scheduler` will be equivalent to
+                `DPM++2M` in stable-diffusion-webui. On top of that, setting this option to True will make it `DPM++2M
+                Karras`.
```
Comment (sayakpaul) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion_k_diffusion.py:464]: This is how docstrings should be I think. 

Could you maybe just add a link to the paper that introduced Karras sigmas?

**Review by sayakpaul — APPROVED** [review]
I love everything about this PR. 

Changes introduced, PR description, and the considerations given to it. 

Well done!

**Comment by alexisrolland** [comment]
@takuma104 out of curiosity, why was this functionality implemented in separate pipeline than StableDiffusionPipeline rather than as a scheduler for StableDiffusionPipeline?

**Comment by takuma104** [comment]
@alexisrolland Currently, there is a `use_karras_sigmas=True` option in `DPMSolverMultistepScheduler` (#3001), so please use it if you want to use `karras_sigmas` in a pipeline other than this one.

**Comment by asthinasthi** [comment]
I was looking for all the available samplers. The list is here https://github.com/crowsonkb/k-diffusion/blob/master/k_diffusion/sampling.py
```
sample_dpm_fast
sample_dpm_adaptive
sample_dpmpp_2s_ancestral
sample_dpmpp_sde
sample_dpmpp_2m
sample_dpmpp_2m_sde
```
In case anyone is interested.
